fdbserver/workloads/RocksCycle.actor.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+ * RocksCycle.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "flow/Arena.h"
+#include "flow/IRandom.h"
+#include "flow/Trace.h"
+#include "flow/WipedString.h"
+#include "flow/serialize.h"
+#include "fdbrpc/simulator.h"
+#include "fdbrpc/TokenSign.h"
+#include "fdbrpc/TenantInfo.h"
+#include "fdbclient/FDBOptions.g.h"
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbserver/TesterInterface.actor.h"
+#include "fdbserver/workloads/workloads.actor.h"
+#include "fdbserver/workloads/BulkSetup.actor.h"
+
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+// This workload is very similar to general CycleWorkload except that we only run it for RocksDB storage engine
+// and turn on knobs such as ROCKSDB_METRICS_IN_SIMULATION. These knobs are known to cause non-determinism, which is
+// why the test sets noUnseed to true. There are cases where we can't pick deterministic random values for rocksdb
+// properties, but still want to exercise the code for test coverage purpose. This workload is meant to solve the test
+// coverage problem while still ensuring all other tests are deterministic.
+struct RocksCycleWorkload : TestWorkload {
```

Comment:
This and the TOML file could be renamed to `Nondeterministic*` instead of `Rocks*`. That way just from the test name, others will expect non-determinism.

Reply:
Done